### ISSUE TITLE
Use github pages to display documentation from PR

### ISSUE THREAD

**Issue by edoddridge** [altMITgcm/MITgcm66h#10]
Reviewing changes to the documentation requires downloading and building the changed files. However, we could set Travis-CI to build the docs and push them to the `gh-pages` branch. From there we can use GitHub pages to serve them as a website.

This would require some clever settings to ensure that people aren't accidentally directed to these docs by search engines, but I think that's possible.

**Comment by jklymak** [comment]
I guess as a potential contributor, I'm not clear how this would work.

If two pull requests are made at the same time, where do the built documents go? Does each pull request get its own entry in `gh-pages`?  If I, as a non-maintainer, issue a pull request, does it make sense that Travis-CI pushes a change to the `gh-pages` branch of the parent repository?  Sounds complicated and risky.  

I'm not sure that this case is any different than any other code-review.  If I submit a change to the code base the only way to properly check that it works is to compile and run it locally.  Reviews are often made with some element of trust that the compiled version does what its supposed to.  

If a PR wants to be particularly helpful, they could link to a locally-built version of the page on their own webserver that contians their changes. i.e. http://web.uvic.ca/~jklymak/testMITdocs/

**Comment by jklymak** [comment]
Just to follow up a bit, another advantage of building your own docs is that you can use [sphinx-autobuild](https://github.com/GaretJax/sphinx-autobuild) to automatically refresh the docs in a browser window.  Not quite WYSIWYG, but it does update quite quickly.

**Comment by edoddridge** [comment]
I agree @jklymak - this issue was opened following discussion of possible ways to streamline the review process. 

The `gh-pages` branch would only be able to store the built documentation from the most recent push to *any* pull request, so it would become very complicated if there were multiple pull requests open.

Having a contributor build and locally host the html output is definitely the high achiever way to deal with reviewing changes to the documentation.
